### ISSUE TITLE
No Data despite data existing in the buckets

### ISSUE THREAD

**Issue by AlexLJordan** [ActivityWatch/activitywatch#998]
- [x] I am on the [latest](https://github.com/ActivityWatch/activitywatch/releases/latest) ActivityWatch version.
 - [x] I have searched the issues of this repo and believe that this is not a duplicate.

- **OS name and version**: NixOS 23.05
- **ActivityWatch version**: 0.12.2 (rust)

## Describe the bug

I'm missing data for a about 14 days. 

I checked the JSON of my buckets: They do contain a bunch of events, which dont look mangled in any way.
The timeline view shows the data properly.

So yeah. Whats happening here?

## To Reproduce

1. have aw-qt in your autostart
2. have modules enabled
3. everything works for months

## Expected behavior

Not losing data.

## Documentation

Today: only half of todays events are tracked:
![Screenshot_20231217_193254](https://github.com/ActivityWatch/activitywatch/assets/10133257/4876aa45-d68f-4b37-9fa9-ac203ceabaf8)

My example will be the 14th december.

No data for previous days:
![no-data](https://github.com/ActivityWatch/activitywatch/assets/10133257/024805ab-2359-4f72-9b9f-8741b172a37f)

Timeline works:
![no-data2](https://github.com/ActivityWatch/activitywatch/assets/10133257/975bd215-0990-4a9e-a349-dfdf2c9c3865)

**Comment by BelKed** [comment]
Could you please try the latest beta version ([`v0.12.3b15`](https://github.com/ActivityWatch/activitywatch/releases/tag/v0.12.3b15)) to verify if the issue still occurs there?

**Comment by The-Hypervisor** [comment]
> Could you please try the latest beta version ([`v0.12.3b15`](https://github.com/ActivityWatch/activitywatch/releases/tag/v0.12.3b15)) to verify if the issue still occurs there?

![Screenshot from 2024-01-16 20-38-08](https://github.com/ActivityWatch/activitywatch/assets/114379310/4a0fc6ed-1f2f-4e8d-823f-5e61e8cd1e67)

**Comment by lucianosarno** [comment]
I have the same issue, and the latest beta version and my OS triggered a security alert.

**Comment by wonka929** [comment]
Look at my comment for this other Issue

https://github.com/ActivityWatch/activitywatch/issues/1006

**Comment by mzfr** [comment]
I have been facing a similar issue. I just noticed that the dashboard for the last 5/6 days just says "No Data". 

I've tried stopping and restarting the server but that didn't help. I even moved from version 0.12.2 to 0.12.3b16 but I still had the same issue.

**OS name and version** - Sonoma 14.2.1
**Current Activity watch version** - 0.12.3b16
   - Same issue on version 0.12.2 as well, under the same OS 

***
<img width="1148" alt="image" src="https://github.com/ActivityWatch/activitywatch/assets/16623935/05bd685b-421e-4f3f-9cfd-6715034faf77">


![image](https://github.com/ActivityWatch/activitywatch/assets/16623935/f5e669a0-5690-4602-b1d8-dc4344a6d01f)

**Comment by mzfr** [comment]
@shoaib281 Nope. For some reason only the Browser Watcher is working so I can only see data in the "browser" tab of the Activity. But other than that the problem is still there for me, even on version *v0.12.3b16*

**Comment by ErikBjare** [comment]
In both @AlexLJordan and @mzfr's timeline screenshots there's missing data. We need data from both aw-watcher-window and aw-watcher-afk to display the Activity view. 

In @AlexLJordan's screenshot there's no aw-watcher-window bucket, so the watcher must have crashed. Simply restarting it would fix it. I'm considering better restart mechanics for watchers that crash to address this.

In @mzfr's screenshot the aw-watcher-afk and aw-watcher-window watchers don't have any overlapping data. Almost like one crashed and then the other was started. This might be a Sonoma problem (I'm still on macOS 13). Not sure what happened there.

**Comment by Rubikoid** [comment]
Faced the same problem after update to `v0.13.1`.
Before update everything was ok.

Checking any previous day also leads to no data.

**OS name and version** - Sonoma 14.3.1.

No data on the main tab:
<img width="1170" alt="Screenshot 2024-06-13 at 18 02 17" src="https://github.com/ActivityWatch/activitywatch/assets/5082689/ce4a69a6-cc14-4385-bbdd-55960fa15276">

Only the editor tab works:
<img width="1161" alt="Screenshot 2024-06-13 at 18 08 31" src="https://github.com/ActivityWatch/activitywatch/assets/5082689/5b23457d-d512-4075-ac24-82349563762c">

But buckets are filling properly:
<img width="1790" alt="Screenshot 2024-06-13 at 18 02 32" src="https://github.com/ActivityWatch/activitywatch/assets/5082689/96348dda-d0e4-447e-9b32-9b2a69e2cf21">

**Comment by Rubikoid** [comment]
Okay, i found a reason.

__sometimes__, in __some__ networks macos can change itself domain to ip/${name}.local.

I had main `rubi-mbp14` AND a `rubi-mbp14.local` buckets.

But due to usage of `find_bucket` in web request (`flood(query_bucket(find_bucket("aw-watcher-window_rubi-mbp14"))); `), instead of `aw-watcher-window_rubi-mbp14 ` was `aw-watcher-window_rubi-mbp14.local` matched

Which, of course, contains no data.
